components/definition/cellRules/NetPyNECellRules.js

```diff
@@ -361,179 +418,207 @@ export default class NetPyNECellRules extends React.Component {
-      }
-      var selectedMechanism = undefined;
-      if ((this.state.selectedMechanism !== undefined) && Object.keys(mechanismsModel).indexOf(this.state.selectedMechanism) > -1) {
-        selectedMechanism = <NetPyNEMechanism cellRule={this.state.selectedCellRule} section={this.state.selectedSection} name={this.state.selectedMechanism} model={mechanismsModel[this.state.selectedMechanism]} />;
-      }
+            <NavigationChevronRight style={styles.rightArrow}/>
 
-      content = (
-        <CardText className={"tabContainer"} expandable={true}>
-          <div className={"thumbnails"}>
-            <div className="breadcrumb">
-              <FloatingActionButton
-                className={"actionButton smallActionButton breadcrumbButton"}
-                style={{ marginTop: "10px", float: "left" }}
-                onClick={() => { that.selectPage("main"); that.setState({ selectedSection: undefined }); }}>
-                {this.state.selectedCellRule}
-              </FloatingActionButton>
-              <div style={{ float: 'left', marginTop: "30px", color: 'grey', fontSize: "20px" }}>&gt;</div>
-              <RaisedButton id={"fromMech2SectionButton"}primary={true} className={"addRectangularButton breadcrumbButton"}
-                onClick={() => { that.selectPage("sections"); that.setState({ selectedMechanism: undefined }); }}
-                style={{ float: 'left', marginTop: "28px", color: 'white' }}
-              >
-                {this.state.selectedSection}
-              </RaisedButton>
-              <div style={{ float: 'left', marginTop: "30px", color: 'grey', fontSize: "20px" }}>&gt;</div>
-              <IconMenu style={{ float: 'left', marginTop: "18px" }}
-                iconButtonElement={
-                  <NetPyNENewMechanism id={"newCellRuleMechButton"} handleClick={this.handleNewMechanism} />
-                }
-                anchorOrigin={{ horizontal: 'left', vertical: 'top' }}
-                targetOrigin={{ horizontal: 'left', vertical: 'top' }}
-              >
-              </IconMenu>
-            </div>
-            <div style={{ clear: "both" }}></div>
-            {mechanisms}
+            <NetPyNENewMechanism
+              blockButton={page != 'mechanisms' && !!model && !!model[selectedCellRule] && !!model[selectedCellRule]['secs'][selectedSection] && Object.keys(model[selectedCellRule]['secs'][selectedSection]['mechs']).length > 0}
+              handleClick={this.handleNewMechanism}
+              handleHierarchyClick={ () => this.handleHierarchyClick('mechanisms')}
+              disabled={selectedSection == undefined || page == 'main'}
+            /> 
           </div>
-          <div className={"details"}>
-            {selectedMechanism}
-            {dialogPop}
-          </div>
-        </CardText>
-      );
-
-    }
+          <div style={{ clear: "both" }}/>
+          { page == "main" ? cellRulesContainer : page == "sections" ? sectionsContainer : mechanismsContainer }
```

Comment:
Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places?